libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java

```diff
@@ -264,16 +264,27 @@ public CredentialIssuerConfig getCriConfig(CriOAuthSessionItem criOAuthSessionIt
     public CredentialIssuerConfig getCriConfigForConnection(String connection, String criId) {
         final String pathTemplate =
                 ConfigurationVariable.CREDENTIAL_ISSUERS.getPath() + "/%s/connections/%s";
-        Map<String, String> result = getSsmParameters(pathTemplate, false, criId, connection);
-
-        if (result == null || result.isEmpty()) {
+        String parameter = getSsmParameter(resolvePath(pathTemplate, criId, connection));
+        if (parameter != null) {
+            try {
+                return objectMapper.readValue(parameter, CredentialIssuerConfig.class);
+            } catch (JsonProcessingException e) {
+                LOGGER.error(
+                        "Failed to parse credential issuer configuration at parameter path {} because: {}",
+                        pathTemplate,
+                        e);
+                throw new RuntimeException(e);
+            }
+        }
+        Map<String, String> parameters = getSsmParameters(pathTemplate, false, criId, connection);
+        if (parameters != null && !parameters.isEmpty()) {
+            return objectMapper.convertValue(parameters, CredentialIssuerConfig.class);
+        } else {
```

Comment:
So the idea is that this will support either config mechanism? Does this mean the two-step deployment actually happens here - i.e. we can remove the 'old' method once the new config is in production?

Probably neater than switching via a temporary config set :)

Reply:
Yes that's the idea:
- push these changes
- switch to new config / params (common-infra PR)
- remove old methods / old config